### DISCUSSION TITLE
[Idea] Extensions Compatible with Current Flarum Version

### DISCUSSION THREAD

**Discussion by DursunCanPoyraz** [rob006-software/flarum-translations#567] [General]
An idea to collaborate with Extiverse? See extensions compatible with Beta 15 in separate list? The translation percentage should be based on the current Flarum version.

current Flarum version extensions: https://extiverse.com/?filter%5Bq%5D=&sort=-created_at&filter%5Bis%5D%5B%5D=&filter%5Btype%5D=

![image](https://user-images.githubusercontent.com/74768208/102927781-a34f2680-44a8-11eb-8735-83197f210350.png)

**Comment by luceos** [comment]
@rob006 already has access to the API and should be able to retrieve most (if not all) necessary information.

**Comment by rob006** [comment]
These widgets with translation progress are generated by Weblate. Unfortunately they have some limitations:
1. They are either per component (extension) or the whole project (all Flarum extensions + core). You can't create widget for subset of extensions.
2. They recognize only two states - either translations are in Weblate and counts in progress, or they don't exist in Weblate and you can't translate them at all. So to remove old extensions from progress, I would need to remove them completely from Weblate. I don't want to do this, because it requires manual work, prevents people from translating extensions that still may be used (some people stuck on beta 13) and generates unnecessary noise for extensions that will be upgraded in near future (I would need to remove it now and readd after few days when they release new version for beta 15). Also there is no reliable way to tell that extension is incompatible with beta 15 - some beta 13 extensions may still work with beta 15 and for example Extiverse may be too aggressive in their requirements.

Two possible solutions would be:
1. Implement custom progress widgets based on data from Weblate API (I'm 99% sure that there is an endpoint for that). **I have no intention to implement this myself**, but still may help with API access and host it on my server.
2. Implementing https://github.com/WeblateOrg/weblate/issues/4089 (and maybe https://github.com/WeblateOrg/weblate/issues/4090) would allow to create custom list of components (extensions) with their own progress status. Then I could create list like "beta 15 compatible" based on some criteria and use widget for this list in README to display progress. I could automate creating lists on our Weblate instance, but the core feature would need to be implemented in Weblate project and I won't be able to do this. There are also [some nuances](https://discuss.flarum.org/d/20807-simplify-translation-process-with-weblate/18) about upgrading Weblate for this project

**Comment by rob006** [comment]
See https://discuss.flarum.org/d/20807-simplify-translation-process-with-weblate/20

While this not affect progress widgets, it should be enough to decide whether extension is worth translating.
